api/v1alpha1/package_webhook.go

```diff
@@ -0,0 +1,206 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/xeipuuv/gojsonschema"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/yaml"
+)
+
+type packageValidator struct {
+	Client  client.Client
+	decoder *admission.Decoder
+}
+
+func InitPackageValidator(mgr ctrl.Manager) error {
+	mgr.GetWebhookServer().
+		Register("/validate-packages-eks-amazonaws-com-v1alpha1-package",
+			&webhook.Admission{Handler: &packageValidator{
+				Client: mgr.GetClient(),
+			}})
+	return nil
+}
+
+func (v *packageValidator) Handle(ctx context.Context, request admission.Request) admission.Response {
+	p := &Package{}
+	err := v.decoder.Decode(request, p)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError,
+			fmt.Errorf("decoding request: %w", err))
+	}
+
+	bundles, err := v.listBundles(ctx)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError,
+			fmt.Errorf("listing package bundles: %w", err))
+	}
+
+	pbc, err := v.getActiveController(ctx)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("getting PackageBundleController: %v", err))
+	}
+
+	activeBundle := v.getActiveBundle(bundles, pbc.Spec.ActiveBundle)
```

Comment:
`getActiveBundle` should really be renamed or reworked to figure out which bundle is active.